chia/wallet/wallet_node.py

```diff
@@ -1539,11 +1539,19 @@ async def validate_block_inclusion(
         return True
 
     async def fetch_puzzle_solution(self, height: uint32, coin: Coin, peer: WSChiaConnection) -> CoinSpend:
-        solution_response = await peer.request_puzzle_solution(
-            wallet_protocol.RequestPuzzleSolution(coin.name(), height)
-        )
-        if solution_response is None or not isinstance(solution_response, wallet_protocol.RespondPuzzleSolution):
-            raise ValueError(f"Was not able to obtain solution {solution_response}")
+        while True:
+            solution_response = await peer.request_puzzle_solution(
+                wallet_protocol.RequestPuzzleSolution(coin.name(), height)
+            )
+            if solution_response is None or not isinstance(solution_response, wallet_protocol.RespondPuzzleSolution):
+                if self.is_trusted(peer):
+                    await asyncio.sleep(0.5)
```

Comment:
shouldn't we stop at some point and tell the user they should fix their full node?